instrumentation/src/javatest/org/oppia/android/instrumentation/player/ExplorationPlayerTest.kt

```diff
@@ -0,0 +1,69 @@
+package org.oppia.android.instrumentation.player
+
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.UiCollection
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.UiScrollable
+import androidx.test.uiautomator.UiSelector
+import androidx.test.uiautomator.Until.hasObject
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.oppia.android.instrumentation.EndToEndTestHelper.startOppiaFromScratch
+
+/** Tests for Explorations. */
+class ExplorationPlayerTest {
+  private val OPPIA_PACKAGE = "org.oppia.android"
+  private val TRANSITION_TIMEOUT = 5000L
+  private lateinit var device: UiDevice
+
+  @Before
+  fun setUp() {
+    // Initialize UiDevice instance
+    device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+    device.startOppiaFromScratch()
+  }
+
+  @Test
+  fun testExploration_prototypeExploration_toolbarTitle_isDisplayedSuccessfully() {
+    navigateToPrototypeExploration()
+    device.wait(
+      hasObject(
+        By.res("$OPPIA_PACKAGE:id/exploration_toolbar_title")
+      ),
+      TRANSITION_TIMEOUT
+    )
+    val explorationTitle = device.findObject(By.res("$OPPIA_PACKAGE:id/exploration_toolbar_title"))
+    assertEquals("Prototype Exploration", explorationTitle.text)
+  }
+
+  /** Navigates and opens the Prototype Exploration using the admin profile. */
+  private fun navigateToPrototypeExploration() {
+    val skip_button = device.findObject(By.res("$OPPIA_PACKAGE:id/skip_text_view"))
+    skip_button?.let {
```

Comment:
After completing the first test case the app doesn't start from the profile chooser activity.
So I came up with this implementation so that we can run multiple tests.

Reply:
Ah, we need to make sure each test is starting from a completely deterministic state otherwise it's very likely the test suites will be flaky (particularly since JUnit may randomize the order the tests are run as a mitigation factor for detecting non-determinism).

To that end, we should be resetting the app state (maybe by closing it & clearing app data during tear-down?) to ensure each state starts in exactly the same place.

Reply:
Still not completed, I think.

Reply:
I've removed the `let` block here.
also mailed you regarding the issues I'm facing regarding clearing app data

Reply:
As discussed over chat, can you please file an issue tracking this specific problem (resetting to a proper state between tests) & open a draft PR with your current WIP solution? I'm fine with this PR getting merged so long as we continue the  necessary work in a follow-up PR.

Reply:
https://github.com/oppia/oppia-android/issues/3693

Created an issue now, will create a pr once I address the changes in other prs

Reply:
https://github.com/oppia/oppia-android/pull/3608/commits/08ffe7596f93ba9d4df801728622447f2ec85a9e
Added this to 2.2 pr so that it wouldn't block 2.3 pr

Reply:
Thanks--considering this resolved then.